kind/pkg/cluster/node.go

```diff
@@ -222,6 +224,12 @@ func (nh *nodeHandle) WriteKubeConfig(dest string) error {
 		buff.WriteString("\n")
 	}
 
+	// create the directory to contain the KUBECONFIG file.
+	err = os.MkdirAll(filepath.Dir(dest), os.ModePerm)
```

Comment:
Quick check: is this the same as kubectl perm wise?

Reply:
Seems sane to me, but we should mimic other tools as much as possible 👍

Reply:
Looks like client-go is using 0755: https://github.com/kubernetes/client-go/blob/5d107d4ebc00ee0ea606ad7e39fd6ce4b0d9bf9e/tools/clientcmd/loader.go#L412

I'll make that change 😄

Reply:
Updated

Reply:
client-go does `0755` https://github.com/kubernetes/client-go/blob/37c3c02ec96533daec0dbda1f39a6b1d68505c79/tools/clientcmd/loader.go#L405-:405